File: ACBrFramework.Net/ACBrFramework.Net/Serial/ECF/Classes/Consumidor.cs
#region COM_INTEROP

#if COM_INTEROP

using System.Runtime.InteropServices;

#endif

#endregion COM_INTEROP

namespace ACBrFramework.ECF
{
	#region COM_INTEROP

#if COM_INTEROP

	[ComVisible(true)]
	[Guid("3DDCFD3E-B4D4-4B44-8D92-45BFA8A48553")]
	[ClassInterface(ClassInterfaceType.AutoDual)]
#endif

	#endregion COM_INTEROP

	public sealed class Consumidor : ACBrComposedComponent
	{
		#region Constructor

		internal Consumidor(ACBrECF acbrECF)
			: base(acbrECF)
		{
		}

		#endregion Constructor

		#region Properties

		public string Documento
		{
			get
			{
				return GetString(ACBrECFInterop.ECF_Consumidor_GetDocumento);
			}
		}

		public string Nome
		{
			get
			{
				return GetString(ACBrECFInterop.ECF_Consumidor_GetNome);
			}
		}

		public string Endereco
		{
			get
			{
				return GetString(ACBrECFInterop.ECF_Consumidor_GetEndereco);
			}
		}

		public bool Atribuido
		{
			get
			{
				return GetBool(ACBrECFInterop.ECF_Consumidor_GetAtribuido);
			}
		}

		public bool Enviado
		{
			get
			{
				return GetBool(ACBrECFInterop.ECF_Consumidor_GetEnviado);
			}
			set
			{
				SetBool(ACBrECFInterop.ECF_Consumidor_SetEnviado, value);
			}
		}

		#endregion Properties

		#region Methods

		public void AtribuiConsumidor(string CPF_CNPJ, string Nome, string Endereco)
		{
			int ret = ACBrECFInterop.ECF_Consumidor_AtribuiConsumidor(this.Handle, ToUTF8(CPF_CNPJ), ToUTF8(Nome), ToUTF8(Endereco));
			CheckResult(ret);
		}

		public void Zera()
		{
			int ret = ACBrECFInterop.ECF_Consumidor_Zera(this.Handle);
			CheckResult(ret);
		}

		#endregion Methods
	}
}

File: jACBrFramework/jACBrFramework/src/jACBrFramework/serial/ecf/Consumidor.java
package jACBrFramework.serial.ecf;

import jACBrFramework.ACBrException;
import jACBrFramework.interop.ACBrECFInterop;
import java.nio.ByteBuffer;

/**
 * Informacoes relativas ao consumidor.
 * 
 * @author Jose Mauro
 * @version Criado em: 26/12/2013 13:44:02, revisao: $Id$
 */
public class Consumidor {
    
    // <editor-fold defaultstate="collapsed" desc="Attributes">    
    /**
     * Instancia ECF para obter o handler.
     */
    ACBrECF ecf = null;
    /**
     * Buffer para leitura dos dados.
     */
    protected final int STR_BUFFER_LEN = 256;
    // </editor-fold>       
    //<editor-fold defaultstate="collapsed" desc="Constructor">
    /**
     * Cria um novo objeto para manipulacao dados da ECF.
     * 
     * @param pEcf instancia para auxiliar a comunicacao com o Acbr.
     */
    Consumidor(ACBrECF pEcf) throws ACBrException {
        this.ecf = pEcf;
    }
    // </editor-fold>    
    //<editor-fold defaultstate="collapsed" desc="Components Methods">   
    /**
     * Obtem o documento de identificacao do consumidor.
     * 
     * @return documento de identificacao do consumidor.
     * @throws ACBrException 
     */
    public String getDocumento() throws ACBrException {
        ByteBuffer lDocumento = ByteBuffer.allocate(STR_BUFFER_LEN);
        int ret = ACBrECFInterop.INSTANCE.ECF_Consumidor_GetDocumento(ecf.getHandle(), lDocumento, STR_BUFFER_LEN);
        ecf.checkResult(ret);   
        return ecf.fromUTF8(lDocumento, ret);
    }

    /**
     * Nome do consumidor.
     * 
     * @return nome do consumidor.
     * @throws ACBrException 
     */
    public String getNome() throws ACBrException {
        ByteBuffer lNome = ByteBuffer.allocate(STR_BUFFER_LEN);
        int ret = ACBrECFInterop.INSTANCE.ECF_Consumidor_GetNome(ecf.getHandle(), lNome, STR_BUFFER_LEN);
        ecf.checkResult(ret);   
        return ecf.fromUTF8(lNome, ret);
    }
    
    /**
     * Endereco do consumidor.
     * 
     * @return endereco do consumidor.
     * @throws ACBrException 
     */
    public String getEndereco() throws ACBrException {
        ByteBuffer lEndereco = ByteBuffer.allocate(STR_BUFFER_LEN);
        int ret = ACBrECFInterop.INSTANCE.ECF_Consumidor_GetEndereco(ecf.getHandle(), lEndereco, STR_BUFFER_LEN);
        ecf.checkResult(ret);   
        return ecf.fromUTF8(lEndereco, ret);
    }  
    
    /**
     * Verifica se houve atribuicao de consumidor.
     * 
     * @return indica se houve atribuicao de consumidor.
     * @throws ACBrException 
     */
    public boolean isAtribuido() throws ACBrException {
        return ACBrECFInterop.INSTANCE.ECF_Consumidor_GetAtribuido(ecf.getHandle()) != 0;
    }   
    
    /**
     * Indica se as informacoes foram enviadas.
     * 
     * @param pEnviado indica que as informacoes foram enviadas.
     * @throws ACBrException 
     */
    public void setEnviado(boolean pEnviado) throws ACBrException {
        int ret = ACBrECFInterop.INSTANCE.ECF_Consumidor_SetEnviado(ecf.getHandle(), pEnviado);
        ecf.checkResult(ret);
    }  
    
    /**
     * Verifica se as informacoes foram enviadas.
     * 
     * @return indica se as informacoes foram enviadas.
     * @throws ACBrException 
     */
    public boolean isEnviado() throws ACBrException {
        return ACBrECFInterop.INSTANCE.ECF_Consumidor_GetEnviado(ecf.getHandle()) != 0;
    }       
    
    /**
     * Realiza a atribuicao das informacoes do consumidor.
     * 
     * @param pCpfCnpj documento de identificacao do consumidor.
     * @param pNome nome do consumidor.
     * @param pEndereco endereco do consumidor.
     * @throws ACBrException 
     */
    public void atribuiConsumidor(String pCpfCnpj, String pNome, String pEndereco) throws ACBrException {
        int ret = ACBrECFInterop.INSTANCE.ECF_Consumidor_AtribuiConsumidor(ecf.getHandle(), 
                ecf.toUTF8(pCpfCnpj), ecf.toUTF8(pNome), ecf.toUTF8(pEndereco));
        ecf.checkResult(ret);
    } 
    
    /**
     * Reinicia as informacoes do consumidor.
     * 
     * @throws ACBrException 
     */
    public void zera() throws ACBrException {
        int ret = ACBrECFInterop.INSTANCE.ECF_Consumidor_Zera(ecf.getHandle());
        ecf.checkResult(ret);
    }    
    // </editor-fold>

}